src/runtime/vm/vm.cc

```diff
@@ -521,13 +603,25 @@ int64_t VirtualMachine::LoadScalarInt(Index r) const {
   return result;
 }
 
-void VirtualMachine::RunLoop() {
+Index VirtualMachine::GetResultRegisterIndex() const {
+  Index op_ind = 0;
+  Instruction instr;
+  // TODO(vvchernov): can it be endless loop?
+  do {
+    instr = code_[op_ind++];
+  } while (instr.op == Opcode::Ret);
```

Comment:
!= Opcode::Ret

Reply:
Thanks very much, it was nasty misprint. About endless loop, we do not know size of code_ array and from one side it potentially does not have return op, from other side it also leads to endless loop in RunLoop method. By default I will think that somebody took care of it.